roody/doxfiles/man/man3/TGTextDialog.3
.TH "TGTextDialog" 3 "Thu Apr 26 2012" "Roody" \" -*- nroff -*-
.ad l
.nh
.SH NAME
TGTextDialog \- 
.SH SYNOPSIS
.br
.PP
.PP
\fC#include <TGTextDialog.h>\fP
.SS "Public Member Functions"

.in +1c
.ti -1c
.RI "\fBTGTextDialog\fP (TGWindow const *, TGWindow const *, UInt_t, UInt_t, char *, char const *, TObjString *, UInt_t=kVerticalFrame)"
.br
.ti -1c
.RI "virtual \fB~TGTextDialog\fP ()"
.br
.ti -1c
.RI "virtual void \fBCloseWindow\fP ()"
.br
.ti -1c
.RI "virtual Bool_t \fBProcessMessage\fP (Long_t, Long_t, Long_t)"
.br
.in -1c
.SS "Protected Attributes"

.in +1c
.ti -1c
.RI "TGCompositeFrame * \fBfF1\fP"
.br
.ti -1c
.RI "TGCompositeFrame * \fBfF2\fP"
.br
.ti -1c
.RI "TGButton * \fBfOkButton\fP"
.br
.ti -1c
.RI "TGButton * \fBfCancelButton\fP"
.br
.ti -1c
.RI "TGLayoutHints * \fBfL1\fP"
.br
.ti -1c
.RI "TGLayoutHints * \fBfL5\fP"
.br
.ti -1c
.RI "TGLayoutHints * \fBfL6\fP"
.br
.ti -1c
.RI "TGLayoutHints * \fBfL21\fP"
.br
.ti -1c
.RI "TGTextEntry * \fBfText\fP"
.br
.ti -1c
.RI "TGTextBuffer * \fBfBLabel\fP"
.br
.ti -1c
.RI "TGLabel * \fBfLabel\fP"
.br
.ti -1c
.RI "TObjString * \fBfRetObjStr\fP"
.br
.in -1c
.SH "Detailed Description"
.PP 
Definition at line 35 of file TGTextDialog.h.
.SH "Constructor & Destructor Documentation"
.PP 
.SS "TGTextDialog::TGTextDialog (TGWindow const *, TGWindow const *, UInt_t, UInt_t, char *, char const *, TObjString *, UInt_t = \fCkVerticalFrame\fP)"
.SS "TGTextDialog::~TGTextDialog ()\fC [virtual]\fP"
.PP
Definition at line 101 of file TGTextDialog.cxx.
.SH "Member Function Documentation"
.PP 
.SS "void TGTextDialog::CloseWindow ()\fC [virtual]\fP"
.PP
Definition at line 112 of file TGTextDialog.cxx.
.PP
Referenced by ProcessMessage().
.SS "Bool_t TGTextDialog::ProcessMessage (Long_tmsg, Long_tparm1, Long_tdummy)\fC [virtual]\fP"
.PP
Definition at line 120 of file TGTextDialog.cxx.
.SH "Member Data Documentation"
.PP 
.SS "TGTextBuffer* \fBTGTextDialog::fBLabel\fP\fC [protected]\fP"
.PP
Definition at line 43 of file TGTextDialog.h.
.PP
Referenced by ProcessMessage().
.SS "TGButton* \fBTGTextDialog::fCancelButton\fP\fC [protected]\fP"
.PP
Definition at line 40 of file TGTextDialog.h.
.PP
Referenced by ~TGTextDialog().
.SS "TGCompositeFrame* \fBTGTextDialog::fF1\fP\fC [protected]\fP"
.PP
Definition at line 38 of file TGTextDialog.h.
.PP
Referenced by ~TGTextDialog().
.SS "TGCompositeFrame * \fBTGTextDialog::fF2\fP\fC [protected]\fP"
.PP
Definition at line 38 of file TGTextDialog.h.
.PP
Referenced by ~TGTextDialog().
.SS "TGLayoutHints* \fBTGTextDialog::fL1\fP\fC [protected]\fP"
.PP
Definition at line 41 of file TGTextDialog.h.
.PP
Referenced by ~TGTextDialog().
.SS "TGLayoutHints * \fBTGTextDialog::fL21\fP\fC [protected]\fP"
.PP
Definition at line 41 of file TGTextDialog.h.
.PP
Referenced by ~TGTextDialog().
.SS "TGLayoutHints * \fBTGTextDialog::fL5\fP\fC [protected]\fP"
.PP
Definition at line 41 of file TGTextDialog.h.
.PP
Referenced by ~TGTextDialog().
.SS "TGLayoutHints * \fBTGTextDialog::fL6\fP\fC [protected]\fP"
.PP
Definition at line 41 of file TGTextDialog.h.
.PP
Referenced by ~TGTextDialog().
.SS "TGLabel* \fBTGTextDialog::fLabel\fP\fC [protected]\fP"
.PP
Definition at line 44 of file TGTextDialog.h.
.PP
Referenced by ~TGTextDialog().
.SS "TGButton* \fBTGTextDialog::fOkButton\fP\fC [protected]\fP"
.PP
Definition at line 39 of file TGTextDialog.h.
.PP
Referenced by ProcessMessage(), and ~TGTextDialog().
.SS "TObjString* \fBTGTextDialog::fRetObjStr\fP\fC [protected]\fP"
.PP
Definition at line 45 of file TGTextDialog.h.
.PP
Referenced by ProcessMessage().
.SS "TGTextEntry* \fBTGTextDialog::fText\fP\fC [protected]\fP"
.PP
Definition at line 42 of file TGTextDialog.h.
.PP
Referenced by ProcessMessage(), and ~TGTextDialog().

.SH "Author"
.PP 
Generated automatically by Doxygen for Roody from the source code.


roody/src/TGTextDialog.cxx
/*********************************************************************

  Name:         TGTextDialog.cxx
  Created by:   Stephan Ritt
  $Id: TGTextDialog.cxx 169 2006-09-26 22:34:44Z olchansk $
  
*********************************************************************/
//////////////////////////////////////////////////////////////////////
// This Code is written by Stephan Ritt                             //
// found in rmidas.c, a component of the MIDAS                      //
// (see http://midas.triumf.ca/)                                    //
//////////////////////////////////////////////////////////////////////

#include "TGTextDialog.h"

ClassImp(TGTextDialog)

TGTextDialog::TGTextDialog( TGWindow const *p, const TGWindow *main,
                            UInt_t w, UInt_t h, char *label, char const *preLoad,
                            TObjString *ret_str,
                            UInt_t options)
    : TGTransientFrame(p, main, w, h, options)
{
  // Create a dialog to enter a single line text entry
  //
  fRetObjStr = ret_str;
  //
  ChangeOptions( (GetOptions()&~kVerticalFrame)|kHorizontalFrame );
  //
  fF1 = new TGCompositeFrame( this, 60, 20, kVerticalFrame | kFixedWidth );
  fF2 = new TGCompositeFrame( this, 60, 20, kHorizontalFrame );
  //
  fOkButton = new TGTextButton( fF1, new TGHotString("Ok"), 1 );
  fCancelButton = new TGTextButton( fF1, new TGHotString("Cancel"), 2 );
  fF1->Resize( fOkButton->GetDefaultWidth()+40, GetDefaultHeight() );
  //
  fOkButton->Associate(this);
  fCancelButton->Associate(this);
  //
  fL1 = new TGLayoutHints( kLHintsTop | kLHintsExpandX, 2, 2, 3, 0 );
  fL21 = new TGLayoutHints( kLHintsTop | kLHintsRight, 2, 5, 10, 0 );
  //
  fF1->AddFrame( fOkButton, fL1 );
  fF1->AddFrame( fCancelButton, fL1 );
  AddFrame( fF1, fL21 );
  //
  fLabel = new TGLabel( fF2, new TGHotString(label) );
  //
  fBLabel = new TGTextBuffer(200);
  fOkButton->SetState(kButtonDisabled);
  //
  fText = new TGTextEntry( fF2, fBLabel );
  fText->Associate(this);
  fText->Resize( 220, fText->GetDefaultHeight() );
  fText->SelectAll();
  if( strlen(preLoad) )fText->SetText( preLoad );
  //
  fL5 = new TGLayoutHints( kLHintsLeft|kLHintsCenterY, 3, 5, 0, 0 );
  fL6 = new TGLayoutHints( kLHintsLeft|kLHintsCenterY, 0, 2, 0, 0 );
  //
  fF2->AddFrame( fLabel, fL5 );
  fF2->AddFrame( fText, fL5 );
  AddFrame( fF2, fL1 );
  //
  MapSubwindows();
  Resize( GetDefaultSize() );
  //
  // position relative to the parent's window
  Int_t ax, ay;
  if( main )
  {
    Window_t wdum;
    gVirtualX->TranslateCoordinates(
     main->GetId(), GetParent()->GetId(),
     (Int_t)(((TGFrame *)main)->GetWidth()-fWidth)>>1,
     (Int_t)(((TGFrame *)main)->GetHeight()-fHeight)>>1,
     ax, ay, wdum );
  }
  else
  {
    UInt_t root_w, root_h;
    gVirtualX->GetWindowSize( fClient->GetRoot()->GetId(), ax, ay,
                              root_w, root_h );
    ax = (root_w-fWidth) >> 1;
    ay = (root_h-fHeight) >> 1;
  }
  //
  Move( ax, ay );
  SetWMPosition( ax, ay );
  //
  SetWindowName("Enter Text");
  //
  SetMWMHints( kMWMDecorAll|kMWMDecorMaximize|kMWMDecorMenu,
               kMWMFuncAll|kMWMFuncMaximize|kMWMFuncResize,
               kMWMInputModeless );
  //
  MapWindow();
  fClient->WaitFor(this);
}

TGTextDialog::~TGTextDialog()
{
  // Clean up text dialog
  delete fCancelButton;
  delete fOkButton;
  delete fText;
  delete fLabel;
  delete fF1; delete fF2;
  delete fL1; delete fL5; delete fL6; delete fL21;
}

void TGTextDialog::CloseWindow()
{
  // Close the dialog.
  // On close the dialog will be deleted and cannot be re-used.
  //
  DeleteWindow();
}

Bool_t TGTextDialog::ProcessMessage( Long_t msg, Long_t parm1, Long_t dummy )
{
  switch (GET_MSG(msg))
  {
    case kC_COMMAND:
    {
      switch (GET_SUBMSG(msg))
      {
        case kCM_BUTTON:
        {
          switch (parm1)
          {
            case 1:
            {
              fRetObjStr->SetString(fText->GetText());
              CloseWindow();
              break;
            }
            case 2:
            {
              fRetObjStr->SetString("-1");
              CloseWindow();
              break;
            }
          }
          break;
        }
        default:
        {
          break;
        }
      }
      break;  
    }
    case kC_TEXTENTRY:
    {
      switch (GET_SUBMSG(msg))
      {
        case kTE_TEXTCHANGED:
        {
          char const *string = fBLabel->GetString();
          strlen(string)==0 ? fOkButton->SetState(kButtonDisabled) :
                              fOkButton->SetState(kButtonUp);
          break;
        }
        case kTE_ENTER:
        {
          char const *string = fBLabel->GetString();
          if( strlen(string) != 0 )
          {
            fRetObjStr->SetString(fText->GetText());
            CloseWindow();
          }
          break;
        }
        default:
        {
          break;
        }
      }
      break;
    }
    default:
    {
      break;
    }
  }
  return kTRUE;
}
